vendors/marvell/boards/mw300_rd/ports/wifi/iot_wifi.c

```diff
@@ -209,18 +210,28 @@ WIFIReturnCode_t WIFI_ConnectAP( const WIFINetworkParams_t * const pxNetworkPara
     strncpy(home_nw.name, "aws_network", WLAN_NETWORK_NAME_MAX_LENGTH);
 
     /* Set SSID as passed by the user */
-    strncpy(home_nw.ssid, pxNetworkParams->pcSSID, IEEEtypes_SSID_SIZE + 1);
+    if (pxNetworkParams->ucSSIDLength > 0)
+    {
+        char pcSSID[pxNetworkParams->ucSSIDLength + 1];
+        memcpy(pcSSID, pxNetworkParams->ucSSID, pxNetworkParams->ucSSIDLength);
+        pcSSID[pxNetworkParams->ucSSIDLength] = '\0';
 
-    if(pxNetworkParams->pcPassword) {
+        strlcpy(home_nw.ssid, pcSSID, IEEEtypes_SSID_SIZE + 1);
+    }
+
+    if(pxNetworkParams->xPassword.xWPA.ucLength > 0) {
         /* Set the passphrase */
-        //strncpy(home_nw.security.psk, pxNetworkParams->pcPassword, PASSWORD_MAX_LENGTH);
-        strcpy(home_nw.security.psk, pxNetworkParams->pcPassword);
+        char pcPassphrase[pxNetworkParams->xPassword.xWPA.ucLength + 1];
```

Comment:
Declare static array with predefined size.

Reply:
This is guarded earlier in `WIFI_ConnectAP`. Each of these VLA's, if necessary, should already be size-bounded by similar checks in board's corresponding `iot_wifi.c`. 
`WIFI_ConnectAP` (before VLA instantiation):
```
170     if (!IS_SSID_LEN_VALID(pxNetworkParams->ucSSIDLength) ||
171             (pxNetworkParams->xSecurity != eWiFiSecurityOpen &&
172             !IS_PASSPHRASE_LEN_VALID(pxNetworkParams->xPassword.xWPA.ucLength))) {
173         goto fail;
174     }
```
where
```
143 #define IS_SSID_LEN_VALID(x) \
144         ((x) <=  wificonfigMAX_SSID_LEN && (x) > 0)
```

We could make it static I suppose, but then the RAM usage extends to the life of the program, rather than duration of `WIFI_ConnectAP`.